projects/cashmere/src/lib/stepper/stepper.component.ts

```diff
@@ -80,38 +83,39 @@ export class StepperComponent implements AfterContentInit {
         if (!this.steps) {
             return;
         }
-        if (value < 0 || value >= this.steps.length) {
-            throw Error('The hc-stepper activeIndex value of ' + value + ' is out of bounds');
-        }
-        if (this._routerEnabled) {
+        // if (value !== undefined && (value < 0 || value >= this.steps.length)) {
+        //     throw Error('The hc-stepper activeIndex value of ' + value + ' is out of bounds');
+        // }
+        if (this._routerEnabled && value) {
             this.router.navigate([this.steps[value].routerLink]);
         } else {
             this._activeIndex = value;
             this.activeIndexChange.emit(this._activeIndex);
         }
     }
-    private _activeIndex: number = 0;
+    private _activeIndex = -1;
 
     /** Emits the current zero-based index for the active step whenever it changes */
     @Output()
     activeIndexChange = new EventEmitter<number>();
 
     @HostBinding('class') _hostClass = 'hc-stepper-' + this.color;
 
-    constructor(private router: Router) {}
-
-    ngAfterContentInit() {
-        this._checkForRouterUse();
+    constructor(private router: Router) {
         this.router.events.forEach(event => {
             if (event instanceof NavigationEnd) {
                 const url = event && event.url ? event.url : '';
                 this._findCurrentStep(url);
-                this.activeIndexChange.emit(this._activeIndex);
```

Comment:
I think it should. I didn't test that. Is it not triggering on a route change?

Reply:
Oh - well if we want the emitter to fire on route change, than we probably need to leave line 109 in (looks like it was removed).  Because that should emit any time the router routes.

Reply:
Although I just realized for this to work, `_findCurrentStep()` would have to be allowed to work when activeIndex is anything (not just -1).  Because otherwise line 109 will just emit whatever the last activeIndex was (not what the index is for the new route)

Reply:
This is now working correctly with route changes.